src/Physics/Multinucleon/EventGen/MECGenerator.cxx

```diff
@@ -1333,3 +1303,59 @@ void MECGenerator::LoadConfig(void)
     GetParamDef( "SuSA-MaxXSec-DiffTolerance", fSuSAMaxXSecDiffTolerance, 999999. );
 }
 //___________________________________________________________________________
+double MECGenerator::GetNSVXSecMaxTlctl( const Interaction * in, 
+			   const Range1D_t Tl_range, 
+			   const Range1D_t ctl_range ) const {
+
+  ROOT::Math::Minimizer * min = ROOT::Math::Factory::CreateMinimizer("Minuit2");
+
+  gsl::d2Xsec_dTCosth f(fXSecModel,in) ; 
+  f.SetFactor(-1.); // Make it return negative of cross-section so we can minimize
+  
+  min->SetFunction( f );
+  min->SetMaxFunctionCalls(10000);
+  min->SetTolerance(0.05);
+
+  std::array<string,2> names = { "Tl", "CosThetal" } ;
+  std::array<Range1D_t,2> ranges = { Tl_range, ctl_range } ; 
+
+  std::array<double,2> start, steps, temp_point ;
+  double MinimScanPoints = 20. ; 
```

Comment:
- A bit dangerous increase in complexity. Can we have separate numbers for Tl and Ctl?
- it has to be configurable from xml file